src/com/facebook/buck/android/TrimUberRDotJava.java

```diff
@@ -212,9 +212,12 @@ public Pattern apply(@NonNull String input) {
       // is referenced.  That is a very rare case, though, and not worth the complexity to fix.
       if (m.find()) {
        final String resource = m.group(1);
-       if (!allReferencedResources.contains(packageName + "." + resource) &&
-           (pattern.isPresent() && !pattern.get().matcher(resource).find())) {
-             continue;
+       if (!allReferencedResources.contains(packageName + "." + resource)) {
+         if (pattern.isPresent() && pattern.get().matcher(resource).find()) {
```

Comment:
```
      if (m.find()) {
        final String resource = m.group(1);
        if (!allReferencedResources.contains(packageName + "." + resource) &&
            (!pattern.isPresent() || !pattern.get().matcher(resource).find())) {
          continue;
        }
      }
```
though I personally find inverting it is clearer:
```
        boolean shouldWriteLine = allReferencedResources.contains(packageName + "." + resource)
            || (keepPattern.isPresent() && keepPattern.get().matcher(resource).find());
        if (!shouldWriteLine) {
          continue;
        }
```
![boolean algebra](https://cdn.meme.am/instances/400x/32202866.jpg)